src/tools/rbd_mirror/instance_watcher/Types.h

```diff
@@ -44,7 +44,7 @@ struct ImagePayloadBase : public PayloadBase {
   std::string peer_mirror_uuid;
   std::string peer_image_id;
 
-  ImagePayloadBase() : PayloadBase() {
+  ImagePayloadBase() : PayloadBase(), global_image_id{""}, peer_mirror_uuid{""}, peer_image_id{""} {
```

Comment:
This is another place in ceph where it shows warnings from -Wmaybe-uninitialized. I tried to fix this by giving default values for:

```
  int64_t group_pool;
  string group_id;
  snapid_t snapshot_id;
```

But I was not able to completely remove all the warnings. One still remaining, which was weird.


Warnings:
```
In file included from /home/jcollin/workspace/ceph/src/cls/rbd/cls_rbd_types.cc:5:0:
/home/jcollin/workspace/ceph/src/cls/rbd/cls_rbd_types.h: In member function ‘void cls::rbd::SnapshotNamespaceOnDisk::decode(ceph::buffer::list::iterator&)’:
/home/jcollin/workspace/ceph/src/cls/rbd/cls_rbd_types.h:248:8: warning: ‘<anonymous>.cls::rbd::GroupSnapshotNamespace::group_pool’ may be used uninitialized in this function [-Wmaybe-uninitialized]
 struct GroupSnapshotNamespace {
        ^~~~~~~~~~~~~~~~~~~~~~
/home/jcollin/workspace/ceph/src/cls/rbd/cls_rbd_types.h:248:8: warning: ‘<anonymous>.cls::rbd::GroupSnapshotNamespace::group_pool’ may be used uninitialized in this function [-Wmaybe-uninitialized]
 struct GroupSnapshotNamespace {
        ^~~~~~~~~~~~~~~~~~~~~~
/home/jcollin/workspace/ceph/src/cls/rbd/cls_rbd_types.h:248:8: warning: ‘*((void*)(& temp)+8).cls::rbd::GroupSnapshotNamespace::group_pool’ may be used uninitialized in this function [-Wmaybe-uninitialized]
 struct GroupSnapshotNamespace {
        ^~~~~~~~~~~~~~~~~~~~~~
```